### PR TITLE
To support setting bios for firmware on vsphere upi.

### PR DESCRIPTION
**Pull request by WenXinWei**
add env FIRMWARE_SET to set bios/efi for firmware on vsphere upi.

### CODE REVIEW

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-vsphere-8

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-8

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.11-installer-rehearse-vsphere

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-vsphere

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.11-installer-rehearse-vsphere

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-8

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-vsphere

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-vsphere-8

**Comment by WenXinWei** [comment]
@jiajliu @jinyunma  please help review the pr.

File: ci-operator/step-registry/upi/install/vsphere/upi-install-vsphere-ref.yaml [hunk 1]
```diff
@@ -18,6 +18,10 @@ ref:
     name: ci-route-53
     mount_path: /var/run/vault/vsphere-aws
   env:
+  - name: FIRMWARE_SET
+    default: "efi"
```
Comment (jinyunma) [inline ci-operator/step-registry/upi/install/vsphere/upi-install-vsphere-ref.yaml:22]: suggest to set default value as "", so that we can always use fireware from rhcos base image if env FIRMWARE_SET is not set.

File: ci-operator/step-registry/cucushift/installer/rehearse/vsphere/upi/firmware-bios/provision/cucushift-installer-rehearse-vsphere-upi-firmware-bios-provision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+chain:
+  as: cucushift-installer-rehearse-vsphere-upi-firmware-bios-provision
```
Comment (jinyunma) [inline ci-operator/step-registry/cucushift/installer/rehearse/vsphere/upi/firmware-bios/provision/cucushift-installer-rehearse-vsphere-upi-firmware-bios-provision-chain.yaml:2]: compared with chain `cucushift-installer-rehearse-vsphere-upi-provision`, seems no difference.

And I see that you define ENV `FIRMWARE_SET` in workflow, how about to move ENV on job config level? then we don't need to add new workflow/provision chains

File: ci-operator/step-registry/upi/install/vsphere/upi-install-vsphere-ref.yaml (hunk 1 above)
Comment (WenXinWei) [reply ci-operator/step-registry/upi/install/vsphere/upi-install-vsphere-ref.yaml:22]: ok.got it.

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-8

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-8

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml [hunk 3]
```diff
@@ -1888,12 +1888,13 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-vsphere-upi-disconnected-secureboot
-- as: vsphere-upi-p3-f14
-  cron: 32 3 5,20 * *
+- as: vsphere-upi-firmware-bios-p3-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1891]: how about rename the job name to `biso` instead of `firmware-bios` for a shorter name.

Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1891]: ok. let me update

**Comment by jianlinliu** [comment]
@liangxia pls help review, thanks.

**Comment by jianlinliu** [comment]
@rvanderp3 can you help review this PR?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 4]
```diff
@@ -3347,6 +3347,17 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-vsphere-upi-encrypt
+- as: vsphere-upi-bios-p3-f14
+  cron: 54 5 8,23 * *
+  steps:
+    cluster_profile: vsphere
+    env:
+      E2E_RUN_TAGS: '@amd64 and @vsphere-upi and @network-openshiftsdn and not @fips'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:3355]: Here should be `@network-ovnkubernetes`

Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:3355]: Thank you for your reminder.

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-8

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml [hunk 5]
```diff
@@ -1888,12 +1888,13 @@ tests:
...
     workflow: cucushift-installer-rehearse-vsphere-upi-disconnected-secureboot
-- as: vsphere-upi-p3-f14
-  cron: 32 3 5,20 * *
+- as: vsphere-upi-bios-p3-f14
+  cron: 32 2 7,22 * *
   steps:
     cluster_profile: vsphere
     env:
-      E2E_RUN_TAGS: '@amd64 and @vsphere-upi and @network-openshiftsdn and not @fips'
+      E2E_RUN_TAGS: '@amd64 and @vsphere-upi and @network-ovnkubernetes and not @fips'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1896]: We should revert the change on this line.  As in 4.11 and earlier, network defaults to SDN, while in 4.12 and above, the network defaults to OVN.

**Comment by liangxia** [comment]
Looks like a rebase is needed.

**Comment by liangxia** [comment]
Since it conflicts again and again, I would suggest break the PR into two PRs. One PR with changes in `ci-operator/step-registry/upi/install/vsphere/`, and the other one for the test changes.

**Comment by WenXinWei** [comment]
> Since it conflicts again and again, I would suggest break the PR into two PRs. One PR with changes in `ci-operator/step-registry/upi/install/vsphere/`, and the other one for the test changes.

seems no conflict this time.

**Comment by jianlinliu** [comment]
Per the discussion in https://redhat-internal.slack.com/archives/C999USB0D/p1689599284075119?thread_ts=1643748103.305689&cid=C999USB0D, close it.
